integrations/sonarqube/main.py

```diff
@@ -35,11 +101,33 @@ async def on_project_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
             "No projects found in Sonarqube, failing the resync to avoid data loss"
         )
 
+    if not fetched_projects:
+        logger.error("No projects found in Sonarqube")
+        raise RuntimeError(
+            "No projects found in Sonarqube, failing the resync to avoid data loss"
+        )
+
 
 @ocean.on_resync(ObjectKind.ISSUES)
 async def on_issues_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    selector = cast(SonarQubeIssueResourceConfig, event.resource_config).selector
+    query_params = selector.generate_request_params()
+    project_query_params = produce_project_params(selector.project_api_filters)
+    initial_component_query_params = (
+        selector.project_api_filters.generate_request_params()
+        if selector.project_api_filters
+        else {}
+    )
+    component_params = produce_component_params(
+        sonar_client, selector, initial_component_query_params
+    )
```

Comment:
it will be great if you have a centralized way of generating the project query params based on whether we are using internal or GA API. At the moment, it looks like we are duplicating this logic

Reply:
The `produce_component_params` method is used for generating query params when using the internal API.  The other function is when using the GA API